### PR TITLE
Feature/365 validation engine

### PR DESCRIPTION
**Pull request by IvanSheyenko**

### CODE REVIEW

File: src/SIM.Sitecore9Installer/Validation/ValidationFactory.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    private static ValidationFactory _instance;
+
+    internal ValidationFactory()
+    {
+      JObject doc =
+        JObject.Parse(File.ReadAllText(Path.Combine(Directory.GetCurrentDirectory(),
+          "GlobalParamsConfig/Validators.json")));
+      List<ValidatorDefinition> validatorDefinitions =
+        doc["ValidatorDefinitions"].ToObject<List<ValidatorDefinition>>();
+      foreach (var definition in validatorDefinitions)
```
Comment (ViacheslavAfanasyev) [inline src/SIM.Sitecore9Installer/Validation/ValidationFactory.cs]: please use explicit type

File: src/SIM.Sitecore9Installer/Validation/ValidationFactory.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace SIM.Sitecore9Installer.Validation
+{
+  public class ValidationFactory
+  {
+    private  Dictionary<string, IValidator> validators=new Dictionary<string, IValidator>();
```
Comment (ViacheslavAfanasyev) [inline src/SIM.Sitecore9Installer/Validation/ValidationFactory.cs]: move to ctor

Comment (IvanSheyenko) [reply src/SIM.Sitecore9Installer/Validation/ValidationFactory.cs]: done

File: src/SIM.Sitecore9Installer/Validation/ValidationFactory.cs (hunk 1 above)
Comment (IvanSheyenko) [reply src/SIM.Sitecore9Installer/Validation/ValidationFactory.cs]: done

**Pull request merged**